components/job-orchestration/job_orchestration/scheduler/query/query_scheduler.py

```diff
@@ -527,12 +627,11 @@ def handle_pending_query_jobs(
                     pending_search_jobs.append(new_search_job)
                 active_jobs[job_id] = new_search_job
 
-            elif QueryJobType.EXTRACT_IR == job_type:
-                extract_ir_config = ExtractIrJobConfig.parse_obj(msgpack.unpackb(job_config))
-                archive_id, file_split_id = get_archive_and_file_split_ids_for_extraction(
-                    db_conn, extract_ir_config
+            elif job_type in [QueryJobType.EXTRACT_IR, QueryJobType.EXTRACT_JSON]:
+                archive_id, target_id = get_archive_and_target_ids_for_stream_extraction(
```

Comment:
Can we create a local (within this file) abstract class with methods for each of the 5 operations necessary to be differentiated between IR and JSON stream extractions? Then we can create an implementation for IR and one for JSON so that essentially the abstract class mechanism takes care of the if IR: ... elif JSON: ... for us.

I haven't thought this through, but I have a feeling you could potentially even hide more of the stream-specific logic in this class (e.g., you probably wouldn't need to return target_id).